doc/hotspot-style.md

```diff
@@ -463,8 +463,9 @@ worthwhile, given the alternatives.
 ### Memory Allocation
 
 Do not use the standard global allocation and deallocation functions
-(operator new and related functions).  Use of these functions by HotSpot
-code is disabled for some platforms.
+(operator new and related functions). Typically, uses of the global
+operator new are inadvertent and therefore often associated with memory
+leaks. Use of these functions by HotSpot code is disabled for some platforms.
```

Comment:
First, it's not exactly a new sentence, just one moved from elsewhere in our code base (from a file that was deleted in the companion PR to this one).

Second, it is true; we have seen problems in the (distant) past of exactly the form claimed.  The problem is that HotSpot is an irregular user of C++, including via assembly code and tortuous stack frame manipulation (deopt handlers etc.).  It's easy to accidentally emit a use of of global `op new` through ten layers of C++ header file, and in HotSpot it's also easy to break the careful matching of constructors to destructors that C++ relies on.  The result is a storage leak.

Kim, I could see you thinking, also, that this sort of observation doesn't belong in a style guide, and a lot of these nuggets might tend to bloat which obscures the useful parts of the style guide.  (An over-long guide is not a useful guide after all.)  You might suggest where this rationale information goes, if not here.  But I think it fits well enough here.  And if it isn't inserted here, or some other new place, it will be lost because of David's file deletion in the other PR related to this one.  I don't want it to get lost.

Reply:
I would also like that sentence to not be lost.

Reply:
Would this not be something for a footnote or an addendum?

Reply:
If that sentence (or something like it) were at the end of the following "rationale" paragraph (where I think @rose00 originally suggested) then I'd be okay with it, as that would put it in the scope of the NMT requirement.  Maybe even remove "typically" in that location.

Reply:
Moved to the rationale as suggested.